### DISCUSSION TITLE
can not remote control archlinux kde 5.27.4 wayland

### DISCUSSION THREAD

**Discussion by Silentdoer** [rustdesk/rustdesk#4045] [General]
error: Failed to create capturer

**Comment by rustdesk** [comment]
Could you help dive into and figure it out?

**Comment by Silentdoer** [comment]
after: yay -Sy gst-plugins-base gst-plugin-pipewire
it works

**Reply by rustdesk** [reply]
@fufesou add these to arch deps. and also for the other distros (may have different names).

**Reply by rustdesk** [reply]
@Silentdoer Thanks, good job.

**Reply by THEGOLDENPRO** [reply]
Thank you, this fixed it for me too. 🙏

**Reply by ChenZhongPu** [reply]
Before installing `gst-plugins-base gst-plugin-pipewire`, it also reported 

> error: Failed to create capturer

After installing them, what I got is only is screen glitch. Any idea to solve this?

<img width="1212" alt="image" src="https://github.com/rustdesk/rustdesk/assets/2310958/1abb11e8-c269-437f-81a3-25c70c3d82c6">

**Reply by rustdesk** [reply]
Please try out https://github.com/rustdesk/rustdesk/releases/tag/nightly on both sides

**Reply by ChenZhongPu** [reply]
> Please try out https://github.com/rustdesk/rustdesk/releases/tag/nightly on both sides

I tried 1.2.5 preview, and it works well generally. But the shared screen is NOT as smooth as before.

**Reply by ChenZhongPu** [reply]
I also tried the nightly build. It has more bugs because we cannot specify which screen to share if two monitors are set up.

**Reply by fufesou** [reply]
You can clear the selection here. https://github.com/rustdesk/rustdesk/pull/7988

But if you select a second display, you will connect to the primary display when you reconnect. We have to wait the upstream to confirm and fix it.

https://github.com/flatpak/xdg-desktop-portal/issues/1371

**Reply by rustdesk** [reply]
We  removed specifying monitor in nigthly, because it crashes on many Wayland.

**Reply by ChenZhongPu** [reply]
Now I am using the latest 1.2.6 on both sides (Mac, and Linxu with Wayland). It works well to connect to Mac, but it still reports "[failed to](error: Failed to create capturer) " while connecting to Linux from Mac.

I also tried nightly built, but with no luck. 

I think it may have something to do with my dual monitor setup, because it also shows the prompt like "please select the monitor to share" (it disappears in a very short time).

**Reply by rustdesk** [reply]
@ChenZhongPu could you try out this https://github.com/rustdesk/rustdesk/issues/8600#issuecomment-2207953508?

**Reply by ChenZhongPu** [reply]
> @ChenZhongPu could you try out this [#8600 (comment)](https://github.com/rustdesk/rustdesk/issues/8600#issuecomment-2207953508)?

I tried:

> ❯ ./xcap_monitor_capture_wl
thread 'main' panicked at examples/monitor_capture.rs:15:35:
called `Result::unwrap()` on an `Err` value: XcbConnError(ClosedParseErr)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

> ❯ RUST_BACKTRACE=1 ./xcap_monitor_capture_wl
thread 'main' panicked at examples/monitor_capture.rs:15:35:
called `Result::unwrap()` on an `Err` value: XcbConnError(ClosedParseErr)
stack backtrace:
   0: rust_begin_unwind
   1: core::panicking::panic_fmt
   2: core::result::unwrap_failed
   3: monitor_capture::main

**Reply by rustdesk** [reply]
@ChenZhongPu Thanks, @21pages will contact you and debug together with you.

**Reply by ChenZhongPu** [reply]
@21pages  Sorry, I didn't get the point. Which `server` do you mean?

**Reply by rustdesk** [reply]
@ChenZhongPu ignore that message about `server`, please check your email [chenloveit@gmail.com](mailto:chenloveit@gmail.com), @21pages will contact you.
